infinispan-spring-boot-samples/session/src/main/resources/application.properties

```diff
@@ -0,0 +1,24 @@
+infinispan.remote.server-list=127.0.0.1:11222
+infinispan.remote.client-intelligence=BASIC
+
+# near caching
+infinispan.remote.near-cache-mode=INVALIDATED
+infinispan.remote.near-cache-max-entries=100
+
+# activates statistics for actuator
+infinispan.remote.statistics=true
+infinispan.remote.jmx=true
+
+management.endpoints.web.exposure.include=*
+
+# Security settings. Comment these properties if you are using a non secured server
+infinispan.remote.use-auth=true
+infinispan.remote.sasl-mechanism=PLAIN
+infinispan.remote.auth-realm=default
+infinispan.remote.auth-server-name=infinispan
+infinispan.remote.auth-username=Titus Bramble
+infinispan.remote.auth-password=Shambles
+
+# Important to add the java marshaller if you need SpringSession or avoid Protostream
+infinispan.remote.marshaller=org.infinispan.commons.marshall.JavaSerializationMarshaller
+#infinispan.remote.java-serial-whitelist=java.util.HashMap
```

Comment:
@ryanemerson to make it work I need to add this .... (a cache names "sessions" needs to exist in the server)

Reply:
Strange, it's added as part of the default [whitelist](https://github.com/infinispan/infinispan/blob/9f093155e486bd5479912f2987a791ed3d906c19/spring/spring5/spring5-remote/src/main/java/org/infinispan/spring/remote/AbstractRemoteCacheManagerFactory.java#L36) and seems Ok in our Spring boot integration test ... I'm guessing `AbstractRemoteCacheManagerFactory::initializeMarshallingProperties` is not being called for some reason.

Reply:
The issue with our integration test. `RemoteSpringSessionTest`, is that it programatically sets the whitelist on the SpringRemoteCacheManagerFactoryBean using `AbstractRemoteCacheManagerFactory.SPRING_JAVA_SERIAL_WHITELIST`. We need to add an additional integration test that utilises properties based configuration, without `java-serial-whitelist`, as well as fixing the spring integration logic so that the `SPRING_JAVA_SERIAL_WHITELIST` classes are automatically configured if omitted.